### DISCUSSION TITLE
Does Doctrine ORM suffer of the problem of useless order by on some ORM prefetches ?

### DISCUSSION THREAD

**Discussion by LLyaudet** [doctrine/doctrine2#11375] [Support Questions] [answered]
Hello,

Recently I found a problem with Django and submitted a patch in a PR:
https://code.djangoproject.com/ticket/35309
https://github.com/django/django/pull/17984
I also developped a work-around for old versions of Django:
https://github.com/LLyaudet/django-monkey-patches/blob/main/src/django_monkey_patches/django__orm__prefetch_without_useless_order_by.py

I don't know yet Doctrine ORM details.
The idea is just that some models can have a default ordering (I assume this functionality is present in Doctrine ORM also).
And that some ORM may still add this ordering in the SQL query whilst it is not needed because only a single object will be prefetched per instance of the original result set.
On very big prefetches I noticed a gain of 10 - 15 %,
since the rest of the DB query is mainly an index scan.
#ClimateChangeBrake

Can someone please dump the queries of such prefetches and see if the bug is present in Doctrine ORM ?
If so submitting a PR should not be too hard.
I will try to help.
But right now, I start by warning the frameworks/ORM I am aware of :)

Best regards,
      Laurent Lyaudet

**Comment by greg0ire** [comment, marked as answer]
I don't think it is affected. In Doctrine ORM, there can be a default order, but it is specified on relationships, and not on entities themselves: https://www.doctrine-project.org/projects/doctrine-orm/en/3.1/reference/attributes-reference.html#attrref_orderby

**Reply by LLyaudet** [reply]
Hello :)
You must be right. It is only documented for ManyToMany and OneToMany and there should be no reason it applies for ManyToOne.
Thanks, best regards,
     Laurent Lyaudet
